src/runtime_src/core/pcie/linux/device_linux.cpp

```diff
@@ -73,6 +74,97 @@ get_pcidev(const xrt_core::device* device)
   return pcidev::get_dev(device->get_device_id(), device->is_userpf());
 }
 
+static query::sdm_sensor_info::result_type
+get_sdm_sensors(const xrt_core::device* device,
```

Comment:
Move this function inside `struct sdm_sensor_info `you have below.
Refactor the function to make it simpler and easier to comprehend; keep the refactored (sub-functions) within the struct.